[cmake][rocRAND] Updated toolchain-linux.cmake to have proper CMAKE_PREFIX_PATH definitions

NguyenNhuDi opened the pull request.

NguyenNhuDi committed:
make rmake compatible with linux

NguyenNhuDi committed:
updated tool chain and cmakeList such that CMAKE_PREFIX_PATH is defined

NguyenNhuDi committed:
implemented default paramaters

NguyenNhuDi committed:
removed CXX default

NguyenNhuDi committed:
updated changelog

NguyenNhuDi committed:
updated changelog to include rmake fix

amd-jnovotny commented on CHANGELOG.md:
```diff
@@ -17,6 +17,11 @@ Documentation for rocRAND is available at
 * `--test|-t` is no longer a required flag for `rtest.py`. Instead, the user can use either `--emulation|-e` or `--test|-t`, but not both.
 * Removed TBB dependency for multi-core processing of host-side generation.
 
+## Fixes
```
```suggestion
## Resolved issues
```

amd-jnovotny commented on CHANGELOG.md:
```diff
@@ -17,6 +17,11 @@ Documentation for rocRAND is available at
+## Fixes
+
+* Fixed an issue where `CMAKE_PREFIX_PATH` was not defined properly in `CMAKELists.txt` and `toolchain-linux.cmake`.
+* Fixed an issue where `cmake_platform_opts` in `rmake.py` was sometimes a string instead of a list.
```
```suggestion
* Fixed an issue in `rmake.py` where `cmake_platform_opts` was sometimes a string instead of a list.
```

amd-jnovotny reviewed (approved):
Changelog looks good overall. Just one section header change and an editorial suggestion.

NguyenNhuDi committed:
Update CHANGELOG.md

Co-authored-by: Jeffrey Novotny <jnovotny@amd.com>

NguyenNhuDi committed:
Update CHANGELOG.md

Co-authored-by: Jeffrey Novotny <jnovotny@amd.com>

NguyenNhuDi committed:
got rid of extra message call

Commit:
removed part of code where the CXX_COMPILER was forced changed to g++ or cl if it was not gnu

NguyenNhuDi committed:
Merge branch 'zenguyen/rmake-remake' into zenguyen/toolchain-remake

NguyenNhuDi commented:
Updated various CMAKE files.

- Moved the position of variable definitions to the top in CMAKELists
- Edited toolchain-linux to define CMAKE_PREFIX_PATH as well
- Removed a part of Dependencies.cmake where it forces the CMAKE_CXX_COMPILER for benchmarks to be cl or g++ if it was not GNU

NguyenNhuDi committed:
updated changelog to include new Dependencies.cmake changes

amd-jnovotny commented on CHANGELOG.md:
```diff
@@ -14,9 +14,15 @@ Documentation for rocRAND is available at
 
 ### Changed
 
+* Removed a section in `cmake/Dependencies.cmake` that was forcing `DCMAKE_CXX_COMPILER` to be either `cl` or `g++` if the compiler was not `GNU`.
```
```suggestion
* Removed a section in `cmake/Dependencies.cmake` that was forcing `DCMAKE_CXX_COMPILER` to be set to either `cl` or `g++` if the compiler was not `GNU`.
```

NguyenNhuDi committed:
Update CHANGELOG.md

Co-authored-by: Jeffrey Novotny <jnovotny@amd.com>

NguyenNhuDi committed:
updated to use google benchmark version 1.8 and cmake version 3.16-3.25

NguyenNhuDi committed:
got rid of run.error_occurred as its no longer supported with google benchmark 1.8
